rdt-open-reader/src/main/java/io/ona/rdt/util/Utils.java

```diff
@@ -212,4 +213,13 @@ public static String getUniqueId(List<UniqueId> uniqueIds) {
 
         return org.smartregister.util.Utils.isEmptyCollection(ids) ? "" : ids.get(0);
     }
+
+    public static boolean isValidJSONObject(String str) {
+        try {
+            new JSONObject(str);
+            return true;
```

Comment:
wish there was a nice way, a little searching brought up this modification, that includes a check for JSONArray, which is valid JSON, but indeed is NOT a valid JSONObject, which you're explicitly checking for here, not sure whether you care about JSONArray, and want to test whether it's JSON in general or only an object, https://stackoverflow.com/questions/10174898/how-to-check-whether-a-given-string-is-valid-json-in-java#10174938

If testing that it's only an object, this looks good as is

Reply:
Agreed, it's super-ugly but seems to be the only option in Java. The `str` is strictly expected to be a `JSONObject` in this case.